src/cargo/core/workspace.rs

```diff
@@ -293,20 +293,27 @@ impl<'cfg> Workspace<'cfg> {
         };
 
         if let Some(list) = members {
-            let root = root_manifest.parent().unwrap();
             for path in list {
+                let root = root_manifest.parent().unwrap();
                 let manifest_path = root.join(path).join("Cargo.toml");
-                self.find_path_deps(&manifest_path)?;
+                self.find_path_deps(&manifest_path, false)?;
             }
         }
 
-        self.find_path_deps(&root_manifest)
+        self.find_path_deps(&root_manifest, false)
     }
 
-    fn find_path_deps(&mut self, manifest_path: &Path) -> CargoResult<()> {
+    fn find_path_deps(&mut self, manifest_path: &Path, is_path_dep: bool) -> CargoResult<()> {
         if self.members.iter().any(|p| p == manifest_path) {
             return Ok(())
         }
+        if is_path_dep
```

Comment:
This check is a bit more complicated that what one would expect. 

Firstly, we want to allow path dependencies outside the wrokspace dir if they have an explicit parent pointer.

Secondly, we want to add explicitelly specified members, even if they are outside and lack a parent pointer. This way we'd bail out with the error in `validate`, instead of simply ignoring a bad member.